src/main/java/spring/intro/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package spring.intro.dao.impl;
+
+import java.util.List;
+import lombok.extern.log4j.Log4j;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import spring.intro.dao.UserDao;
+import spring.intro.model.User;
+
+@Log4j
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public User add(User user) {
+        log.info("Calling method add for " + user);
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            log.info("Insert successful for " + user);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+                log.warn("Transaction rollbacked for user " + user, e);
```

Comment:
better not to swallow exception, but re-throw it, since it's not correct behavior

Reply:
fixed